### ISSUE TITLE
Fix Menu navigation with keyboard and focus handling

### ISSUE THREAD

**Issue by mimarz** [equinor/design-system#605]
blocked by #1186 
We should check if we can solve the focus bug without having the state hook in #1186 finished.

**Describe the bug**

When using arrowkeys (up/down) i should be able to navigate the menu items.

**Steps to reproduce the bug**

1. Trigger open `Menu`
2. Use prop `first` to trigger focus on first/last element
3. 
4.

**Expected behavior**

* First or last menu item should have been focused depending on wether user presses down or up
* MenuItem are not focused when opened by mouseclick

**Specifications**

- Version: 
- Browser: 
- OS: 

**Additional context**

Thorough testing shows that the usage of `useContext` in combination with `useEffect` is somehow screwing up the update and render cycle for the `focusedIndex` prop.

**Comment by oddvernes** [comment]
Fixed by #1671
